src/jit/emitxarch.cpp

```diff
@@ -189,32 +184,17 @@ static bool IsDstSrcImmAvxInstruction(instruction ins)
 #endif // FEATURE_HW_INTRINSICS
 
 // -------------------------------------------------------------------
-// Is4ByteSSE4Instruction: Returns true if the SSE4 instruction
-// is a 4-byte opcode.
+// Is4ByteSSEInstruction: Returns true if the SSE instruction is a 4-byte opcode.
 //
 // Arguments:
 //    ins  -  instruction
 //
 // Note that this should be true for any of the instructions in instrsXArch.h
-// that use the SSE38 or SSE3A macro.
-bool emitter::Is4ByteSSE4Instruction(instruction ins)
+// that use the SSE38 or SSE3A macro but returns false if the VEX encoding is
+// in use, since that encoding does not require an additional byte.
+bool emitter::Is4ByteSSEInstruction(instruction ins)
```

Comment:
Now, the code mixes using `EncodedBySSE38orSSE3A` and `Is4ByteSSEInstruction` that looks confusing. Perhaps, we can rename this function to `EncodedByLegacySSE38orSSE3A` or something similar.

Reply:
I think it might be okay to keep it as is, and instead follow up with the code calling `Is4ByteSSEInstruction` so we can remove this as well.

Currently, the majority of places calling `Is4ByteSSEInstruction` are doing so to determine if they should update the `sz` estimate (7/11 references). This logic would be better suited to be centralized (as most of the other size calculations are).

Of the remaining 4 usages, 2 are `assert(IsAVXInstruction(ins) || Is4ByteSSEInstruction(ins));`, which can be simplified to `assert(IsAVXInstruction(ins) || EncodedBySSE38OrSSE3A(ins));` and the remaining two are for determining if the additional by needs to be output (and could just be simplified to `!UseVEXEncoding() && EncodedBySSE38orSSE3A(ins)` or centralized).

It would be a slightly more involved change, however; so I think it would be better as a followup fix.